### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf3'

### ISSUE THREAD

**Issue by kubco2** [sivel/speedtest-cli#382]
```
# python speedtest.py --list | grep Czech
Traceback (most recent call last):
  File "speedtest.py", line 1453, in <module>
 5148) SychrovNET s.r.o. (Vsetin, Czech Republic) [77.82 km]
 5007) 4M Roznov spol. s r. o. (Roznov, Czech Republic) [84.11 km]
 3394) HVfree.net (Roznov, Czech Republic) [84.11 km]
10855) INTERNEXT 2000 (Zlin, Czech Republic) [85.93 km]
 9585) KAZA.cz s.r.o. (Veseli nad Moravou, Czech Republic) [92.82 km]
 8317) Net-Connect s.r.o. (Hodonin, CZ, Czech Republic) [109.89 km]
    main()
  File "speedtest.py", line 1443, in main
    shell()
  File "speedtest.py", line 1363, in shell
    print_(line)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf3' in position 24: ordinal not in range(128)
```

**Comment by sivel** [comment]
What version of speedtest-cli are you using?  `python speedtest.py --version`

This issue should already be resolved in 1.03

Testing with 1.0.3 I do not get this error.

**Comment by kubco2** [comment]
# python speedtest-cli.py --version
1.0.3
But I tested it on another machine and there it works.

**Comment by kubco2** [comment]
Failing one is on VPS with LANG=C , it works with:

LANG=en_US.UTF-8 python speedtest.py --list

**Comment by sivel** [comment]
Can you confirm what version of python that you are using?  `python -V`

**Comment by kubco2** [comment]
Python 2.7.5 and 3.5.3 both are failing with LANG=C

**Comment by sivel** [comment]
Ok, I'll have to look. I did just realize that my fix in 1.0.3 didn't account for python3, and that I can reproduce with LANG=C, I cannot reproduce on python2.

**Comment by hlx98007** [comment]
PR #357 will fix this bug but apparently this looks not appropriate to the devs.

To reproduce this bug, you need a VPS preferrably located in Asia (which has the non standard characters), git clone speedtest-cli and run `./speedtest-cli.py --list | list`

**Comment by sivel** [comment]
I have made some changes to support python3 in https://raw.githubusercontent.com/sivel/speedtest-cli/51d5b2d2c9f37cac3257123e2c8de38449864a8a/speedtest.py

I however still cannot duplicate this in python2.

In my tests, I used server `11305` which contained the character `ó` (the character referenced in your traceback).

An easy test was simply:

```
LC_ALL=C LANG=C python speedtest.py --server 11305
```

And likewise with python3.

In python2 that succeeds without issue for me, in python3 it failed.  The above referenced file succeeds in both py2 and py3.

**Comment by kubco2** [comment]
Now it works on VPS CentOS7 with Python 2.7.5.
It works on workstation Fedora 25 Python 3.5.3.

It fails on workstation Fedora 25 Python 2.7.13.
```LC_ALL=C LANG=C python2 speedtest.py --list
Traceback (most recent call last):
  File "speedtest.py", line 206, in <module>
    class _Py3Utf8Stdout(TextIOWrapper):
NameError: name 'TextIOWrapper' is not defined```
```LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 python2 speedtest.py --list
Traceback (most recent call last):
  File "speedtest.py", line 206, in <module>
    class _Py3Utf8Stdout(TextIOWrapper):
NameError: name 'TextIOWrapper' is not defined```

**Comment by sivel** [comment]
Thanks for the update, I'll look into the fedora 25 failure.

**Comment by sivel** [comment]
That python2 error on fedora 25 doesn't seem to make sense to me.  That error would indicate that python2 was able to perform `import builtins`, which is not available in python2.  The only time that TextIOWrapper should be called is in the context of python3, as determined by the ability to perform `import builtins`

Can you test the following for me:

```
python2 -c 'import builtins; print(builtins); print(builtins.__file__)'
```

**Comment by kubco2** [comment]
```
python2 -c 'import builtins; print(builtins); print(builtins.__file__);'
<module 'builtins' from '/usr/lib/python2.7/site-packages/builtins/__init__.pyc'>
/usr/lib/python2.7/site-packages/builtins/__init__.pyc
```

```
Python 2.7.13 (default, Jan 12 2017, 17:59:37) 
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import builtins
>>> dir(builtins)
['ArithmeticError', 'AssertionError', 'AttributeError', 'BaseException', 'BufferError', 'BytesWarning', 'DeprecationWarning', 'EOFError', 'Ellipsis', 'EnvironmentError', 'Exception', 'False', 'FloatingPointError', 'FutureWarning', 'GeneratorExit', 'IOError', 'ImportError', 'ImportWarning', 'IndentationError', 'IndexError', 'KeyError', 'KeyboardInterrupt', 'LookupError', 'MemoryError', 'NameError', 'None', 'NotImplemented', 'NotImplementedError', 'OSError', 'OverflowError', 'PendingDeprecationWarning', 'ReferenceError', 'RuntimeError', 'RuntimeWarning', 'StandardError', 'StopIteration', 'SyntaxError', 'SyntaxWarning', 'SystemError', 'SystemExit', 'TabError', 'True', 'TypeError', 'UnboundLocalError', 'UnicodeDecodeError', 'UnicodeEncodeError', 'UnicodeError', 'UnicodeTranslateError', 'UnicodeWarning', 'UserWarning', 'ValueError', 'Warning', 'ZeroDivisionError', '__builtins__', '__doc__', '__file__', '__future_module__', '__name__', '__package__', '__path__', 'abs', 'absolute_import', 'all', 'any', 'apply', 'ascii', 'basestring', 'bin', 'bool', 'buffer', 'bytearray', 'bytes', 'callable', 'chr', 'classmethod', 'cmp', 'coerce', 'compile', 'complex', 'copyright', 'credits', 'delattr', 'dict', 'dir', 'divmod', 'enumerate', 'eval', 'execfile', 'exit', 'file', 'filter', 'float', 'format', 'frozenset', 'getattr', 'globals', 'hasattr', 'hash', 'help', 'hex', 'id', 'input', 'int', 'intern', 'isinstance', 'issubclass', 'iter', 'len', 'license', 'list', 'locals', 'long', 'map', 'max', 'memoryview', 'min', 'next', 'object', 'oct', 'open', 'ord', 'pow', 'print', 'property', 'quit', 'range', 'raw_input', 'reduce', 'reload', 'repr', 'reversed', 'round', 'set', 'setattr', 'slice', 'sorted', 'staticmethod', 'str', 'sum', 'super', 'sys', 'tuple', 'type', 'unichr', 'unicode', 'vars', 'xrange', 'zip']
```


For me it is not as much critical as for VPS was, so I'm fine with current version.

**Comment by sivel** [comment]
My last update should solve the issues for the fedora25 box.  It seems that `python2-future` was the cause, so I had to re-order some import logic to account for that.

https://raw.githubusercontent.com/sivel/speedtest-cli/utf8-encoding/speedtest.py

It should pass on py2 and py3, as well as py2 with python2-future installed

**Comment by kubco2** [comment]
I dont have any errors now

**Comment by itsnotv** [comment]
@sivel I have the same error on Cent OS 7

$ python --version
Python 2.7.5
$ speedtest --version
1.0.3

I can confirm that the below file fixes the error.
https://raw.githubusercontent.com/sivel/speedtest-cli/utf8-encoding/speedtest.py

**Comment by sivel** [comment]
This was resolved in the v1.0.4 release
